assembl/static2/js/app/components/debate/survey/posts.jsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable  react/no-unused-prop-types */
+// @flow
+import React from 'react';
+import { connect } from 'react-redux';
+import { compose, graphql } from 'react-apollo';
+import type { Map } from 'immutable';
+
+import { updateContentLocale } from '../../../actions/contentLocaleActions';
+import withLoadingIndicator from '../../common/withLoadingIndicator';
+import FlatList from '../../common/flatList';
+import Post from './post';
+import PostsQuery from '../../../graphql/QuestionPostsQuery.graphql';
+import { displayAlert } from '../../../utils/utilityManager';
+
+type PostNode = {
+  node: {
+    id: string,
+    originalLocale: string
+  }
+};
+
+type PostsProps = {
+  hasErrors: boolean,
+  posts: {
+    edges: Array<PostNode>
+  },
+  networkStatus: number,
+  defaultContentLocaleMapping: Map,
+  fetchMore: Function,
+  refetch: Function,
+  updateContentLocaleMapping: Function
+};
+
+export class DumbPosts extends React.Component<*, PostsProps, void> {
+  props: PostsProps;
+
+  componentWillMount() {
+    this.updateContentLocaleMappingFromProps(this.props);
+  }
+
+  componentWillReceiveProps(nextProps: PostsProps) {
+    if (nextProps.posts !== this.props.posts) {
+      this.updateContentLocaleMappingFromProps(nextProps);
+    }
+  }
+
+  updateContentLocaleMappingFromProps(props: PostsProps) {
+    const { defaultContentLocaleMapping, updateContentLocaleMapping, posts } = props;
+    const contentLocaleMappingData = {};
+    posts.edges.forEach((edge) => {
+      const post = edge.node;
+      const { id, originalLocale } = post;
+      const contentLocale = defaultContentLocaleMapping.get(originalLocale, originalLocale);
+      contentLocaleMappingData[id] = {
+        contentLocale: contentLocale,
+        originalLocale: originalLocale
+      };
+    });
+
+    updateContentLocaleMapping(contentLocaleMappingData);
+  }
```

Comment:
Is it possible to refactor this updateContentLocaleMappingFromProps logic in a container component so it can be used here and in Survey component?

Reply:
in another ticket ?

Reply:
I will take a look at it. Can you please first rebase from develop?

Reply:
done